launch/client.py

```diff
@@ -744,20 +765,63 @@ def get_async_response(self, async_task_id: str) -> Dict[str, Any]:
         )
         return resp
 
-    def batch_async_request(self, endpoint_id: str, urls: List[str]):
+    def batch_async_request(
+        self,
+        bundle_name: str,
+        urls: List[str],
+        batch_url_file_location: str = None,
+        serialization_format: str = "json",
+    ):
         """
         Sends a batch inference request to the Model Endpoint at endpoint_id, returns a key that can be used to retrieve
         the results of inference at a later time.
 
         Parameters:
-            endpoint_id: The id of the endpoint to make the request to
+            bundle_name: The id of the bundle to make the request to
+            serialization_format: Serialization format of output, either 'pickle' or 'json'.
+                'pickle' corresponds to pickling results + returning
             urls: A list of urls, each pointing to a file containing model input.
                 Must be accessible by Scale Launch, hence urls need to either be public or signedURLs.
+            batch_url_file_location: In self-hosted mode, the input to the batch job will be uploaded
+                to this location if provided. Otherwise, one will be determined from bundle_location_fn()
 
         Returns:
             An id/key that can be used to fetch inference results at a later time
         """
-        raise NotImplementedError
+        f = StringIO()
+        make_batch_input_file(urls, f)
+        f.seek(0)
+
+        if self.self_hosted:
+            # TODO make this not use bundle_location_fn()
```

Comment:
I'm kinda ambivalent about whether to add another `self.register_some_location_fn()`. I think it is possible for people to use `bundle_location_fn` to get a location for the csv, it just gets a bit disorganized, but adding another `register_fn` seems kinda bloated

Reply:
or can we reuse the `bundle_location_fn` but change the function name to be more general?